kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -274,33 +303,107 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
       return null;
     }
 
-    int workerId = nextWorker.getAndIncrement() % workers.size();
-    KafkaConsumerWorker worker = workers.get(workerId);
-    return worker.readRecords(
-        state, timeout, maxBytes,
-        new ConsumerWorkerReadCallback<ClientKeyT, ClientValueT>() {
-          @Override
-          public void onCompletion(
-              List<? extends ConsumerRecord<ClientKeyT, ClientValueT>> records, Exception e
-          ) {
-            updateExpiration(state);
-            if (e != null) {
-              // Ensure caught exceptions are converted to RestExceptions so the user gets a
-              // nice error message. Currently we don't define any more specific errors because
-              // the old consumer interface doesn't classify the errors well like the new
-              // consumer does. When the new consumer is available we may be able to update this
-              // to provide better feedback to the user.
-              Exception responseException = e;
-              if (!(e instanceof RestException)) {
-                responseException = Errors.kafkaErrorException(e);
+    final KafkaConsumerReadTask task = new KafkaConsumerReadTask<KafkaKeyT, KafkaValueT,
+          ClientKeyT, ClientValueT>(
+          state,
+          timeout,
+          maxBytes,
+          new ConsumerWorkerReadCallback<ClientKeyT, ClientValueT>() {
+            @Override
+            public void onCompletion(
+                    List<? extends ConsumerRecord<ClientKeyT, ClientValueT>> records,
+                    Exception e) {
+              KafkaConsumerManager.this.updateExpiration(state);
```

Comment:
It's a little curious that we only update the consumer expiration when the read task completes. I'd have though we'd do it every time the task executes at least.

Reply:
wow that's a very good catch. I did not look into this.